src/VisualStudio.Roslyn.SDK/ComponentDebugger/CapabilityProvider.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem;
+
+namespace Roslyn.ComponentDebugger
+{
+    [Export(ExportContractNames.Scopes.ConfiguredProject, typeof(IProjectCapabilitiesProvider))]
+    [AppliesTo(ProjectCapabilities.CSharp + " | " + ProjectCapabilities.VB)]
```

Comment:
This component will load for every C# and VB project, we need a better approach for this? Is there something unique about these projects?

Reply:
This also loads for Shared Projects.

Reply:
It can apply to any project that has the `<IsRoslyComponent>true</IsRoslynComponent>`. So we load the provider for all projects and add the capability to any that match that. 

Is there some better way to achieve that? I mentioned in the comments we could do it via the targets, but that does require a matching roslyn version.

Reply:
Is there a targets file that it imports we can match on?

Reply:
No, its just a regular old csharp project.

Reply:
> I mentioned in the comments we could do it via the targets, but that does require a matching roslyn version.

Can you expand on what you mean by this? You mean if the project itself has a package reference to a different version of the compiler?

Reply:
Yeah. We could drop the capability provider altogether if we just added the following to the targets file

```xml
<ItemGroup Condition="'$(IsRoslynComponent)' == 'true'">
    <Capability Include="RoslynComponent">
</ItemGroup>
```

But anyone using an older version of the compiler via NuGet wouldn't get the capability, and be able to use it. That might be worth the tradeoff to remove some of the complexity here tho.